Созданы сущности, сервис и репозиторий для приложения типа "оптика"

nataliaezh opened the pull request:
ДЗ к уроку 1

nataliaezh committed:
Созданы сущности, сервис и репозиторий для приложения типа "оптика"

ilya-sukhachev commented on build.gradle:
```diff
@@ -2,13 +2,18 @@ plugins {
 }
 
 dependencies {
+    compile group: 'org.springframework', name: 'spring-core', version: '5.2.11.RELEASE'
```
compile не используем. используй implementation

ilya-sukhachev commented on build.gradle:
```diff
@@ -2,13 +2,18 @@ plugins {
+    compile group: 'org.springframework', name: 'spring-context', version: '5.2.11.RELEASE'
+    compile group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.14.0'
+
     testCompile group: 'junit', name: 'junit', version: '4.12'
```
testCompile не используем используем testImplementation, также junit версии 4 не будет в проекте. можно вообще пока убрать

ilya-sukhachev commented on src/main/java/com/hedgehogs/entities/Glasses.java:
```diff
@@ -0,0 +1,88 @@
+public class Glasses {
+	private Rim rim;
+	private Lenses lenses;
+	private LensesService lensesService;
```
что такое LensesService сервис? доменная модель не может содержать сервис

ilya-sukhachev commented on src/main/java/com/hedgehogs/entities/Glasses.java:
```diff
@@ -0,0 +1,88 @@
+/**
+ * @author natalya_ezhkova@mail.ru
+ */
+public class Glasses {
```
нет описания. необходимо все задачи сдавать в соответствии с требованиями

ilya-sukhachev commented on src/main/java/com/hedgehogs/entities/Rim.java:
```diff
@@ -0,0 +1,10 @@
+ * @author natalya_ezhkova@mail.ru
+ */
+public class Rim {
+	public void choosing() {
```
такие заглушки недопустимы

ilya-sukhachev commented on src/main/java/com/hedgehogs/entities/Sunglasses.java:
```diff
@@ -0,0 +1,7 @@
+/**
+ * @author natalya_ezhkova@mail.ru
+ */
+public class Sunglasses {
```
что за класс, почему нет полей?

ilya-sukhachev commented on src/main/java/com/hedgehogs/services/LensesService.java:
```diff
@@ -0,0 +1,46 @@
+/**
+ * @author natalya_ezhkova@mail.ru
+ */
+public class LensesService {
```
необходимо реализовать crud для всех элементов доменной модели

ilya-sukhachev commented on src/main/java/com/hedgehogs/Launcher.java:
```diff
@@ -0,0 +1,30 @@
+	public static void main(String[] args) {
+		ApplicationContext context = new ClassPathXmlApplicationContext("config.xml");
+		Glasses glasses = context.getBean("glasses", Glasses.class);
+		Lenses lenses  = context.getBean("lenses", Lenses.class);
```
Примеры использования должны быть показаны для всех crud операций

nataliaezh committed:
В build.gradle compile изменено на implementation
Из доменной модели удалено поле, относящееся к бизнес-слою
Добавлено описание задачи для класса Очки
Добавлены поля в класс Rim
Добавлены поля в класс Sunglasses
Добавлено описание для каждого класса
Добавлена реализация методов CRUD

ilya-sukhachev commented on src/main/java/com/hedgehogs/entities/Glasses.java:
```diff
@@ -0,0 +1,76 @@
+ */
+
+////Сущность Очки, описывает очки, которые будут изготовлены
+// по индивидуальному заказу в оптике. Реализован метод сборки очков, состоящих из линз и оправы
```
JavaDoc а не комментарий

ilya-sukhachev commented on src/main/java/com/hedgehogs/entities/Glasses.java:
```diff
@@ -0,0 +1,76 @@
+
+public class Glasses {
+	private Rim rim;
+	private Lenses lenses;
```
поля тоже описываются

ilya-sukhachev commented on src/main/resources/config.xml:
```diff
@@ -0,0 +1,27 @@
+			<ref bean="lenses" />
+		</property>
+	</bean>
+	<bean id="glasses" class="com.hedgehogs.entities.Glasses">
```
бины для entity не создаются тут. это только для сервисов.

ilya-sukhachev commented on src/main/java/com/hedgehogs/Launcher.java:
```diff
@@ -0,0 +1,43 @@
+	private static final Logger log  = LogManager.getLogger(Launcher.class.getName());
+	public static void main(String[] args) {
+		ApplicationContext context = new ClassPathXmlApplicationContext("config.xml");
+		Glasses glasses = context.getBean("glasses", Glasses.class);
```
объекты создаешь сама. а для автосвязывания используешь сервисы.

nataliaezh committed:
Добавлены JavaDoc, в конфигурационном файле .xml удалены бины классов-сущностей

nataliaezh committed:
Удалены файлы .html

nataliaezh committed:
Merge remote-tracking branch 'origin/Lesson1' into Lesson1

# Conflicts:
#	src/main/java/com/hedgehogs/Launcher.java
#	src/main/java/com/hedgehogs/repositories/GlassesRepository.java
#	src/main/resources/config.xml
